tests/test_confluence_advanced_mode.py

```diff
@@ -0,0 +1,114 @@
+# coding=utf-8
+import json
+import os
+import tempfile
+import unittest
+from requests import Response
+
+from atlassian import Confluence
+from atlassian.errors import ApiError
+
+
+@unittest.skipIf(
+    not os.path.exists("../credentials.secret"),
+    "credentials.secret missing, skipping test",
+)
+class TestConfluenceAdvancedModeCalls(unittest.TestCase):
+    secret_file = "../credentials.secret"
+
+    """
+        Keep the credentials private, the file is excluded. There is an example for credentials.secret
+        See also: http://www.blacktechdiva.com/hide-api-keys/
+
+        {
+          "host" : "https://localhost:8080",
+          "username" : "john_doe",
+          "password" : "12345678"
+        }
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        try:
+            with open(cls.secret_file) as json_file:
+                credentials = json.load(json_file)
+                cls.confluence = Confluence(
+                    url=credentials["host"],
+                    username=credentials["username"],
+                    password=credentials["password"],
+                )
+        except FileNotFoundError:
+            raise unittest.SkipTest("credentials.secret missing, skipping test")
+        except Exception as err:
+            self.fail("[{0}]: {1}".format(self.secret_file, err))
```

Comment:
Is the surrounding `try` realy needed? If the file can't be read or the connection fails a exception is raised and the test fails.

Reply:
The FileNotFound explicit handling is probably not needed, since it duplicates the skipIf condition for the class.

I left the Exception handling to keep the test similar to the other ones.